File: Hospital Management System/backend/src/main/java/com/hospital/management/service/MedicineService.java
package com.hospital.management.service;

import com.hospital.management.model.Medicine;

import java.util.List;
import java.util.Optional;

public interface MedicineService {
    
    Medicine createMedicine(Medicine medicine);
    
    List<Medicine> getAllMedicines();
    
    Optional<Medicine> getMedicineById(Long id);
    
    Medicine updateMedicine(Long id, Medicine medicineDetails);
    
    void deleteMedicine(Long id);
}

File: Hospital Management System/frontend/src/services/medicineService.js
import api from './api';

const medicineService = {
  getAllMedicines: async () => {
    try {
      const response = await api.get('/medicines');
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  addMedicine: async (medicine) => {
    try {
      const response = await api.post('/medicines', medicine);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  getMedicineById: async (id) => {
    try {
      const response = await api.get(`/medicines/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  updateMedicine: async (id, medicine) => {
    try {
      const response = await api.put(`/medicines/${id}`, medicine);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  deleteMedicine: async (id) => {
    try {
      const response = await api.delete(`/medicines/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },
};

export default medicineService;

